eng/scripts/Pack-Crates.ps1

```diff
@@ -126,26 +126,65 @@ function Get-PackagesToBuild() {
   return $buildOrder
 }
 
-function Initialize-VendorDirectory() {
-  $path = "$RepoRoot/target/vendor"
-  Invoke-LoggedCommand "cargo vendor $path" -GroupOutput | Out-Host
-  return $path
-}
+# https://doc.rust-lang.org/cargo/reference/registry-web-api.html#publish
+# https://github.com/rust-lang/cargo/blob/5c87c14f9a162daf10d4133fdaab35c72d67b018/crates/crates-io/lib.rs#L42
+function Get-ApiMetadata($package) {
+  $packagePath = Split-Path -Path $package.manifest_path -Parent
+  $readmePath = $package.readme ? (Join-Path -Path $packagePath -ChildPath $package.readme) : $null
+
+  $jsonBody = [ordered]@{
```

Comment:
Why are you copying into exactly the same structure - deps included? Why spend the time and hassle of maintaining code that ends up being the same as just returning the JSON object you want? If you could post process certain fields like `readme` a lot cheaper.

Reply:
They're not the same contract.  One is the metadata handled by cargo internally, the other is the json contract of the crates.io api.  They're just really similar.  I wouldn't want to deal with cargo adding properties that the API will complain about.

Currently
dependencies -> deps
readme -> readme_file
contents(readme) -> readme
version -> vers

dependencies:
version_req -> req
rename -> explicit_name_in_toml